app/controllers/application_controller.rb

```diff
@@ -264,6 +270,8 @@ def access_denied(options ={})
       destination = options[:redirect].blank? ? user_path(current_user) : options[:redirect]
       flash[:error] = ts "Sorry, you don't have permission to access the page you were trying to reach."
       redirect_to destination
+    elsif logged_in_as_admin? && options[:admin_access_denied]
```

Comment:
Can you remove this `elsif` now that `rescue_from Pundit::NotAuthorizedError` no longer calls `access_denied`? Same for the options parameter of `admin_only_access_denied`.